<a href="https://colab.research.google.com/github/kompallyGyaneshwar215/DS-Lab/blob/main/Lab%206.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Week 6: Implement & Analyze exploratory data analysis techniques using statistical
measures


1.	Measures of Central Tendency

In [ ]:
import pandas as pd
from sklearn.datasets import load_iris
from statistics import mode

iris = load_iris()
df = pd.DataFrame(iris.data, columns=iris.feature_names)
print("First 5 rows of dataset:")
print(df.head())
# Mean (average)
mean_values = df.mean()
print("\nMean values:")
print(mean_values)
# Median (middle value)
median_values = df.median()
print("\nMedian values:")
print(median_values)
# Mode (most frequent value)

mode_values = df.mode().iloc[0]
print("\nMode values:")
print(mode_values)


First 5 rows of dataset:
   sepal length (cm)  sepal width (cm)  petal length (cm)  petal width (cm)
0                5.1               3.5                1.4               0.2
1                4.9               3.0                1.4               0.2
2                4.7               3.2                1.3               0.2
3                4.6               3.1                1.5               0.2
4                5.0               3.6                1.4               0.2

Mean values:
sepal length (cm)    5.843333
sepal width (cm)     3.057333
petal length (cm)    3.758000
petal width (cm)     1.199333
dtype: float64

Median values:
sepal length (cm)    5.80
sepal width (cm)     3.00
petal length (cm)    4.35
petal width (cm)     1.30
dtype: float64

Mode values:
sepal length (cm)    5.0
sepal width (cm)     3.0
petal length (cm)    1.4
petal width (cm)     0.2
Name: 0, dtype: float64


2. Measures of Dispersion

In [ ]:
import pandas as pd
from sklearn.datasets import load_iris

iris = load_iris()
df = pd.DataFrame(iris.data, columns=iris.feature_names)
print("First 5 rows of dataset:")
print(df.head())
# Range (max - min)
range_values = df.max() - df.min()
print("\nRange values:")
print(range_values)
# Variance
variance_values = df.var()
print("\nVariance values:")
print(variance_values)
# Standard Deviation
std_values = df.std()
print("\nStandard Deviation values:")
print(std_values)
# Interquartile Range (IQR)
Q1 = df.quantile(0.25)
Q3 = df.quantile(0.75)
IQR = Q3 - Q1
print("\nInterquartile Range (IQR):")
print(IQR)


First 5 rows of dataset:
   sepal length (cm)  sepal width (cm)  petal length (cm)  petal width (cm)
0                5.1               3.5                1.4               0.2
1                4.9               3.0                1.4               0.2
2                4.7               3.2                1.3               0.2
3                4.6               3.1                1.5               0.2
4                5.0               3.6                1.4               0.2

Range values:
sepal length (cm)    3.6
sepal width (cm)     2.4
petal length (cm)    5.9
petal width (cm)     2.4
dtype: float64

Variance values:
sepal length (cm)    0.685694
sepal width (cm)     0.189979
petal length (cm)    3.116278
petal width (cm)     0.581006
dtype: float64

Standard Deviation values:
sepal length (cm)    0.828066
sepal width (cm)     0.435866
petal length (cm)    1.765298
petal width (cm)     0.762238
dtype: float64

Interquartile Range (IQR):
sepal length (cm)    1.3
sepal width (cm)    

3. Shape of Distribution

In [ ]:
import pandas as pd
from sklearn.datasets import load_iris

iris = load_iris()
df = pd.DataFrame(iris.data, columns=iris.feature_names)
print("First 5 rows of dataset:")
print(df.head())
# Skewness (asymmetry of distribution)
skewness_values = df.skew()
print("\nSkewness values:")
print(skewness_values)
# Kurtosis (peakedness / tail heaviness)
kurtosis_values = df.kurt()
print("\nKurtosis values:")
print(kurtosis_values)


First 5 rows of dataset:
   sepal length (cm)  sepal width (cm)  petal length (cm)  petal width (cm)
0                5.1               3.5                1.4               0.2
1                4.9               3.0                1.4               0.2
2                4.7               3.2                1.3               0.2
3                4.6               3.1                1.5               0.2
4                5.0               3.6                1.4               0.2

Skewness values:
sepal length (cm)    0.314911
sepal width (cm)     0.318966
petal length (cm)   -0.274884
petal width (cm)    -0.102967
dtype: float64

Kurtosis values:
sepal length (cm)   -0.552064
sepal width (cm)     0.228249
petal length (cm)   -1.402103
petal width (cm)    -1.340604
dtype: float64


4. Correlation & Association

•	Pearson Correlation: Linear relationship between two continuous variables.
•	Spearman Rank Correlation: Non-linear monotonic relationships.
•	Chi-Square Test: Association between categorical variables.


5. Hypothesis Testing
•	T-tests: Compare means between groups.
•	ANOVA (Analysis of Variance): Compare means across multiple groups.
•	Z-tests: Compare sample mean to population mean


6. Outlier Detection
•	Z-score Method: Flag values beyond ±3 standard deviations.
•	IQR Method: Identify values outside 1.5 × IQR range.
•	Grubbs’ Test: Detect single outliers in normally distributed data.
